chapter-13-browser-based-apps/browser-based-application/webpack/webpack.dev.ts

```diff
@@ -21,14 +21,22 @@ import {merge} from 'webpack-merge';
 import baseConfig from './webpack.common.js';
 
 /*
- * A pure SPA experience is used for web development, using a lightweight static server
+ * During development the app uses a lightweight web host and sets strong security headers
+ * Equivalent headers would later be set when the SPA is deployed to a remote web host
  */
-const dirname = process.cwd();
-let devConfig: webpack.Configuration = {
-    mode: 'development',
-};
+let policy = "default-src 'none';";
+policy += " script-src 'self';";
+policy += " connect-src 'self' https://api.webapp.example;";
+policy += " child-src 'self';";
+policy += " img-src 'self';";
+policy += " style-src 'self' https://cdn.jsdelivr.net;";
+policy += " object-src 'none';";
+policy += " frame-ancestors 'none';";
+policy += " base-uri 'self';";
+policy += " form-action 'self'";
 
```

Comment:
Previously I had thought that webpack dev server required insecure options like `unsafe-eval`, but it turns out that it does not, as long as you use source maps. Therefore our example can use secure headers for development, like those in [various online articles](https://aws.amazon.com/blogs/networking-and-content-delivery/adding-http-security-headers-using-lambdaedge-and-amazon-cloudfront/).